src/Engine/Tables/CharacterFrameTable.cpp

```diff
@@ -12,31 +12,21 @@ unsigned int PlayerFrameTable::GetFrameIdByExpression(CharacterExpressionID expr
 }
 
 //----- (00494B10) --------------------------------------------------------
-PlayerFrame *PlayerFrameTable::GetFrameBy_x(unsigned int uFramesetID,
-                                            unsigned int uFrameID) {
-    unsigned int v3;      // esi@1
-    int16_t v6;           // dx@2
-    int v7;               // edx@3
-    char *i;              // eax@3
-    int v9;               // ecx@5
-    PlayerFrame *result;  // eax@6
+PlayerFrame *PlayerFrameTable::GetFrameBy_x(int uFramesetID, int time) {
+    if (this->pFrames[uFramesetID].uFlags & 1 && this->pFrames[uFramesetID].uAnimLength != 0) {
+        // Processing animated character expressions - e.g., CHARACTER_EXPRESSION_YES & CHARACTER_EXPRESSION_NO.
+        int time = (time >> 3) % this->pFrames[uFramesetID].uAnimLength;
 
-    v3 = uFramesetID;
-    if (this->pFrames[uFramesetID].uFlags & 1 &&
-        (v6 = this->pFrames[uFramesetID].uAnimLength) != 0) {
-        v7 = ((signed int)uFrameID >> 3) % (uint16_t)v6;
-        for (i = (char *)&this->pFrames[uFramesetID].uAnimTime;; i += 10) {
-            v9 = *(short *)i;
-            if (v7 <= v9) break;
-            v7 -= v9;
-            ++v3;
+        while (true) {
+            int frameTime = this->pFrames[uFramesetID].uAnimTime;
+            if (time < frameTime)
+                break;
+            time -= frameTime;
+            ++uFramesetID;
```

Comment:
In original code `i` represented anim time of initial frameset inside the loop. And it was not changed.
Here we change frameset every iteration and use it's anim time.
That was a bug?

Reply:
`i` was actually a pointer to `uAnimTime`, and it was advancing inside the loop by `10` on every iteration. So not a bug.

Reply:
(`sizeof(PlayerFrame)` is `10`)

Reply:
Okay yeah, read with my a** it seems)